### PR TITLE
bootstrapping updates

### PR DESCRIPTION
**Pull request by tomkralidis**
- implement tests as pytest
- hookup testing to GitHub Actions CI
- add GitHub Container registry to GitHub Actions CI
- update README with install/run/release workflow
- add flake8 fixes
- cleanup JSON schemas
- update transform workflow to accept byte string (note: needs ecCodes fix)

Note that the jsonschema validation uses a local / patched jsonschema codebase that is able to work with remote references that are YAML.  The schema validation is quite slow (as can be seen in the CI); so some work will be needed to optimize/caching remote schemas.

### CODE REVIEW

File: bufr2geojson/__init__.py [hunk 1]
```diff
@@ -144,20 +134,38 @@ def __init__(self):
...
+
+        with open(csv_code_table) as csvfile:
+            reader = csv.DictReader(csvfile)
+
+            # keep only numeric rows, strip out non numeric rows,
+            # these typically give a range
+            for row in reader:
+                try:
+                    row['CodeFigure'] = int(row['CodeFigure'])
+                    self.code_table[row['FXY']] = row
```
Comment (david-i-berry) [inline bufr2geojson/__init__.py:148]: I think this breaks the code table. There will be multiple rows with the same FXY value, we are only keeping the last.

File: bufr2geojson/__init__.py [hunk 2]
```diff
@@ -521,26 +538,27 @@ def get_code_value(self, fxxyyy: str, code: int) -> str:
...
-        table = self.code_table.loc[(self.code_table["FXY"] == fxxyyy), ]
-        decoded = table.loc[table["CodeFigure"] == code, ]
-        decoded.reset_index(drop=True, inplace=True)
-        if len(decoded) == 1:
-            decoded = decoded.EntryName_en[0]
-        else:
-            assert len(decoded) == 0
-            decoded = None
+
+        table = {k: v for k, v in self.code_table.items() if k == fxxyyy}
```
Comment (david-i-berry) [inline bufr2geojson/__init__.py:542]: table should have multiple rows, I think here we will only have a single row?

File: bufr2geojson/__init__.py [hunk 3]
```diff
@@ -521,26 +538,27 @@ def get_code_value(self, fxxyyy: str, code: int) -> str:
...
+        decoded = None
+        if table[fxxyyy]['CodeFigure'] == code:
```
Comment (david-i-berry) [inline bufr2geojson/__init__.py:544]: Here we need to iterate over the rows to find the one where the CodeFigure matches the code we are looking for.

File: bufr2geojson/__init__.py [hunk 4]
```diff
@@ -59,11 +57,20 @@
...
 THISDIR = os.path.dirname(os.path.realpath(__file__))
-TABLES = f"{THISDIR}{os.sep}resources{os.sep}bufr{os.sep}{BUFR_TABLE_VERSION}"  # noqa
+RESOURCES = f"{THISDIR}{os.sep}resources"
+CODETABLES = {}
+
+ECCODES_DEFINITION_PATH = codes_definition_path()
+if not os.path.exists(ECCODES_DEFINITION_PATH):
+    LOGGER.debug('ecCodes definition path does not exist, trying environment')
+    ECCODES_DEFINITION_PATH = os.environ.get('ECCODES_DEFINITION_PATH')
+    print(ECCODES_DEFINITION_PATH)
```
Comment (david-i-berry) [inline bufr2geojson/__init__.py:69]: LOGGER.debug() rather than print?

File: requirements.txt [hunk 5]
```diff
@@ -1,7 +1,8 @@
 cfunits
 click
 eccodes
+jsonschema @ git+https://github.com/tomkralidis/jsonschema@ref-yaml#egg=jsonschema
 jsonschema
```
Comment (david-i-berry) [inline requirements.txt:5]: delete?

File: bufr2geojson/__init__.py (hunk 4 above)
Comment (tomkralidis) [reply bufr2geojson/__init__.py]: fixed

File: requirements.txt (hunk 5 above)
Comment (tomkralidis) [reply requirements.txt]: fixed

**Pull request merged**